components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authzservermetadata/AuthzServerMetadataEndpoint.java

```diff
@@ -0,0 +1,109 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.endpoint.authzservermetadata;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.base.MultitenantConstants;
+import org.wso2.carbon.base.ServerConfigurationException;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.discovery.OIDCDiscoveryEndPointException;
+import org.wso2.carbon.identity.discovery.OIDCProcessor;
+import org.wso2.carbon.identity.oauth.common.OAuthConstants;
+import org.wso2.carbon.identity.oauth.endpoint.util.factory.OIDCProviderServiceFactory;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+
+/**
+ * Rest implementation of OAuth 2 Authorization Server Metadata endpoint.
+ */
+@Path("/{issuer}/.well-known/oauth-authorization-server")
+public class AuthzServerMetadataEndpoint {
+
+    private static final Log log = LogFactory.getLog(AuthzServerMetadataEndpoint.class);
+    private static final String AUTHZ_SERVER_METADATA_ENDPOINT_PATH_COMPONENT_VALUE_TOKEN = "token";
+
+    @GET
+    @Produces("application/json")
+    public Response getAuthzServerMetadata(
+            @PathParam("issuer") String authzServerMetadataEpPathComponent, @Context HttpServletRequest request) {
+
+        String tenantDomain = null;
+        Object tenantObj = IdentityUtil.threadLocalProperties.get().get(OAuthConstants.TENANT_NAME_FROM_CONTEXT);
+        if (tenantObj != null) {
+            tenantDomain = (String) tenantObj;
+        }
+        if (StringUtils.isEmpty(tenantDomain)) {
+            tenantDomain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+        }
+        if (isValidIssuer(authzServerMetadataEpPathComponent)) {
+            return this.getResponse(request, tenantDomain);
+        } else {
+            Response.ResponseBuilder errorResponse = Response.status(HttpServletResponse.SC_BAD_REQUEST);
+            if (log.isDebugEnabled()) {
+                log.debug("The oauth authorization server metadata path component is " + authzServerMetadataEpPathComponent +
+                        " . The expected oauth authorization server metadata path component is either '"
```

Comment:
Log doesn't seem to make sense with the constant appended

Reply:
Removed the redundant logs in [0f2b08f](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2794/commits/0f2b08f5be3bf45c11c19b2257e399a1239a9add)